### ISSUE TITLE
about the train

### ISSUE THREAD

**Issue by Usernamezhx** [breadbread1984/R3DCNN#1]
first of all. thanks your cool code. about the net architecture,why not use the softmax after the rnn model? thanks in advance

**Comment by breadbread1984** [comment]
[ctc_loss](https://tensorflow.google.cn/api_docs/python/tf/nn/ctc_loss) requires input be the logits.

**Comment by panxuezm** [comment]
Hi.When I use your code to train in depth image, I met a problem that the loss drops very slowly and oscillatorily. Finally the accuracy of training set is about 90% and the accuracy of testing set is about 50%.Do you know what the problem might be?Thanks in advance

**Comment by breadbread1984** [comment]
the reported accuracy of the model trained solely on depth image is around 83%

**Comment by Usernamezhx** [comment]
sorry to bother you again. when I run the train__r3dcnn.py . it will show me that : 
![_20180906134327](https://user-images.githubusercontent.com/14010625/45137365-e3584580-b1da-11e8-95e4-d8188c3caeec.jpg)
![_20180906134327](https://user-images.githubusercontent.com/14010625/45137400-0daa0300-b1db-11e8-8a63-467e1800b5e0.jpg)
I have installed the tensorflow version 1.4.0. I think this may be about the  SparseTensor problem.  [https://github.com/tensorflow/tensorflow/issues/13766](url)
My label sparseTensor looks like this : 
`SparseTensorValue(indices=array([[0]], dtype=int64), values=array([6]), dense_shape=array([1], dtype=int64))`
as you see my label in int six

**Comment by Usernamezhx** [comment]
I have solved it by update my tensorflow to version 1.5.0...........

**Comment by Usernamezhx** [comment]
sorry. ` InvalidArgumentError (see above for traceback): Input to reshape is a tensor with 12042240 values, but the requested shape has 3010560.` i change the batch size get the same erro . and I  set the single nvidia card get the same erro. when i load my data: data.get_shape().as_list() show that : [?, 10,8,112,112,3]

**Comment by Usernamezhx** [comment]
yeah, solved . change the code from `clips = tf.decode_raw(feature['clips'],out_type = tf.uint8)` to `clips = tf.decode_raw(feature['clips'],out_type = tf.int32)`

**Comment by castiel520** [comment]
> Hi.When I use your code to train in depth image, I met a problem that the loss drops very slowly and oscillatorily. Finally the accuracy of training set is about 90% and the accuracy of testing set is about 50%.Do you know what the problem might be?Thanks in advance

@panxuezm @breadbread1984 Hi guys, I managed to train the network 50000 steps, but i'm confused about how to get the train/test accuracy, here is my eval output:
{'loss': 0.04681665, 'global_step': 290862, 'mean_edit_distance': 0.0}

**Comment by breadbread1984** [comment]
you can modify the returned estimator in eval mode to print the metric of accuracy.

**Comment by castiel520** [comment]
@breadbread1984 I manage to get it work and the final accuracy is around 0.28，thanks, I wonder the best accuracy you got on the dataset?

**Comment by breadbread1984** [comment]
would you try the model I shared at the link on README.md?


在 2018-11-02 18:09:24，"Castiel" <notifications@github.com> 写道：


@breadbread1984 I manage to get it work and the final accuracy is around 0.28，thanks, I wonder the best accuracy you got on the test set?

**Comment by castiel520** [comment]
@breadbread1984 yes, i tried your C3D model https://github.com/breadbread1984/c3d and download from baidu cloud.

**Comment by sainisanjay** [comment]
@castiel520 @breadbread1984 May i know how to download pre-trained C3D Model from from baidu cloud. When i tried it asked Chinese phone number for registration. Any idea how to download???

**Comment by breadbread1984** [comment]
would you suggest a file sharing platform other than dropbox, google drive which I can't access from my country. I will try to share the file with it.

**Comment by sainisanjay** [comment]
Thanks for swift response, can you share on my iCloud; sanjuair@icloud.com

**Comment by breadbread1984** [comment]
i dont use apple, you could train c3d from scratch. it takes just a day on 1080ti to converge.-------- 原始邮件 --------主题：Re: [breadbread1984/R3DCNN] about the train (#1)发件人：Sanjay Saini 收件人：breadbread1984/R3DCNN 抄送：Konquer your Mind ,Mention Thanks for swift response, can you share on my iCloud; sanjuair@icloud.com

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/breadbread1984/R3DCNN","title":"breadbread1984/R3DCNN","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/breadbread1984/R3DCNN"}},"updates":{"snippets":[{"icon":"PERSON","message":"@sainisanjay in #1: Thanks for swift response, can you share on my iCloud; sanjuair@icloud.com"}],"action":{"name":"View Issue","url":"https://github.com/breadbread1984/R3DCNN/issues/1#issuecomment-449540376"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/breadbread1984/R3DCNN/issues/1#issuecomment-449540376",
"url": "https://github.com/breadbread1984/R3DCNN/issues/1#issuecomment-449540376",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by sainisanjay** [comment]
@breadbread1984, as I don’t have sufficient computing power thus its difficult for me to train C3D model from scratch. Just please let me know which file sharing channel do you use other than Baidu cloud.

**Comment by anguszxd** [comment]
@castiel520 hello, it seems that you've gotten a pretty good result as you reported the eval result : {'loss': 0.04681665, 'global_step': 290862, 'mean_edit_distance': 0.0} But how's the eval accuracy ? You said it to be around 0.28, it is unmatched to such a low loss ....
I'm also trainin this model. The training loss is very low, but the eval loss is high. My eval result is {'loss': 6.3208323, 'accuracy': 0.2695, 'mean_edit_distance': 0.758, 'global_step': 166929} . I'm not sure this training process is all right....

**Comment by MengXinChengXuYuan** [comment]
@castiel520  Hi did you solve the problem ?

**Comment by buaa-luzhi** [comment]
@anguszxd
Hello, what changes did you make to get the loss value to be so low. Did you test the performance of the model offline.
Thanks so much!
